compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/SkikoTestOwner.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import androidx.compose.ui.InternalComposeUiApi
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.semantics.SemanticsNode
+
+/**
+ * Marker interface to be implemented by the Skiko `TestOwner`. The purpose is to allow code in the
+ * `ui-test` module to access functions specific to the Skiko test implementation (e.g. in
+ * `ui-test-junit4`).
+ */
+// Ideally, this would be @VisibleForTesting, but that's an Android annotation we don't want a
+// dependency on
+@InternalComposeUiApi
+interface SkikoTestOwner {
```

Comment:
Looking at the code, it seems the more right approach would be to move this interface and `runSkikoComposeUiTest` into `ui-test` module. It will be needed anyway when we try getting rid of junit.

This way we don't need to introduce any new public API's.

Will it be possible to do, or postpone introducing `captureToImage` for non-junit tests? Can we have non-junit tests right now?

Reply:
We already have captureToImage for non-junit tests. This is needed to add it for junit-based (rule) tests.

Reply:
> We already have captureToImage for non-junit tests

Do we have support of non-junit tests, but its API still in `ui-junit` module?

Reply:
My point was if that is the case, the right place for such API is `ui-test`. And if it is in `ui-test`, we only need `internal SkikoTestOwner`, not `public SkikoTestOwner`.

Reply:
`runSkikoComposeUiTest` exposes `SkikoComposeUiTest` which has several dependencies on `ui-test-junit4`:
- The `expect interface ComposeUiTest` (`@ExperimentalTestApi`)
- AbstractMainTestClock (internal)
- UncaughtExceptionHandler (internal)
- `runOnUiThread` and `isOnUiThread` (internal)

We could make it expose an interface instead, but it should still extend `ComposeUiTest`, no?

So I guess we could move it, but with coordination with Google.

Reply:
There was a refactoring of ui-test some time ago - splitting it into `ui-test` and `ui-test-junit4`, so we can test Compose without using `ui-test-junit` library. It seems isn't finished, and we have still some junit-independent functionality in `ui-test-junit`.

If we can't move only our code to `ui-test`, let's proceed the easy way, as in your code, but just move the interface as described [here](https://github.com/JetBrains/compose-multiplatform-core/pull/589#discussion_r1214678047).